### PR TITLE
Animated WebP support

### PR DESCRIPTION
**Pull request by Aely0**
Adds support for animated WebP images. Used the already existing GIF implementation as a template for most of it.

* [x] I have followed the instructions in the PR template

### CODE REVIEW

File: crates/egui/src/widgets/image.rs [hunk 1]
```diff
@@ -796,57 +798,90 @@ pub fn paint_texture_at(
...
+pub fn has_webp_header(bytes: &[u8]) -> bool {
+    bytes.len() >= 12 && &bytes[0..4] == b"RIFF" && &bytes[8..12] == b"WEBP"
+}
+
+fn is_animated_image_uri(uri: &str) -> bool {
+    is_gif_uri(uri) || is_webp_uri(uri)
+}
+
+fn are_animated_image_bytes(uri: &str, bytes: &[u8]) -> bool {
+    (is_gif_uri(uri) && has_gif_magic_header(bytes)) || (is_webp_uri(uri) && has_webp_header(bytes))
```
Comment (emilk) [inline crates/egui/src/widgets/image.rs:886]: I think we should only check the bytes here; the uri may or may not contain the file ending, but if the magic bytes are correct it should always be fine

**Review by emilk — APPROVED** [review]
Looks good, if tested

File: crates/egui/src/widgets/image.rs (hunk 1 above)
Comment (Aely0) [reply crates/egui/src/widgets/image.rs:886]: I'm fine with that

**Comment by Aely0** [comment]
Needed this feature for a project I'm working on, currently using this branch as a dependency. So as far as testing goes, everything has been working as expected. All the quirks I came across were ironed out pre-PR.

File: examples/images/src/cat.webp
[the remark this answers is not in the document]
Comment (emilk) [reply examples/images/src/cat.webp:1]: I tried to fix this for you, but failed:

```
Pushing to github.com:Aely0/egui
error: Authentication error: Authentication required: You must have push access to verify locks
error: failed to push some refs to 'github.com:Aely0/egui'
```

You need to run this:

```
git lfs track examples/images/screenshot.png
git lfs track examples/images/src/cat.webp
git checkout .gitattributes
git add -u
git commit -m "Move images onto git lfs"
git push
```

[the remark this answers is not in the document]
Comment (Aely0) [reply examples/images/src/cat.webp:1]: Done, though strange why you were not able to push to this branch
image
I thought that was the whole point of this checkbox

[the remark this answers is not in the document]
Comment (emilk) [reply examples/images/src/cat.webp:1]: Weird 🤔

[the remark this answers is not in the document]
Comment (emilk) [reply examples/images/src/cat.webp:1]: This seems to be a limitation of `git lfs`:

https://github.com/cli/cli/discussions/8794

**Comment by emilk** [comment]
`examples/images/screenshot.png` is not on `git lfs`, which is why the CI is red.

We should add better instructions for how to solve this problem to the CI output and to `CONTRIBUTING.md`…

**Comment by Aely0** [comment]
> `examples/images/screenshot.png` is not on `git lfs`, which is why the CI is red.

Is it not what [`38e36a8`](https://github.com/emilk/egui/pull/5470/commits/38e36a8b4409477794d7e584e073f12fb5e04379) was supposed to do? Or is there something else I still need to do from my end?

**Comment by Aely0** [comment]
Oh, this part, "If the CI complains about this, make sure you run git add --renormalize .."

**Pull request merged**